proxy/src/fs_watch.rs

```diff
@@ -48,29 +48,22 @@ where
 
     Interval::new(Instant::now(), interval)
         .map_err(|e| error!("timer error: {:?}", e))
-        .filter_map(move |_| {
+        .filter(move |_| {
+            let mut any_changes = false;
             for file in &files {
-                match file.has_changed() {
-                    Ok(true) => {
-                        trace!("{:?} changed", &file.path);
-                        return Some(());
-                    }
-                    Ok(false) => {
-                        // If the hash hasn't changed, keep going.
-                    }
-                    Err(ref e) if e.kind() == io::ErrorKind::NotFound => {
-                        // A file not found error indicates that the file
-                        // has been deleted.
-                        trace!("{:?} deleted", &file.path);
-                        return Some(());
-                    }
-                    Err(ref e) => {
+                let has_changed = file
+                    .has_changed()
+                    .unwrap_or_else(|e| {
                         warn!("error hashing {:?}: {}", &file.path, e);
-                    }
+                        false
+                    });
+                if has_changed {
+                    any_changes = true;
```

Comment:
can't we just break the loop at this point?  If so, i'd probably extract this into another fn so you can just return early.

Reply:
@olix0r: that's what we were doing previously, but it was causing false positives when more than one file has changed since the last poll. if we return early here, we won't have hashed all the files yet, so we'll see more "changes" on the next poll even if there haven't been any new changes since the last poll.